packages/in_app_purchase_tizen/tizen/src/billing_manager.cc

```diff
@@ -0,0 +1,264 @@
+#include "billing_manager.h"
+
+#include <flutter/method_channel.h>
+#include <flutter/standard_method_codec.h>
+
+#include <cassert>
+#include <iomanip>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "log.h"
+
+static server_type ConvertServerType(const char *server_type_string) {
+  if (strcasecmp("DEV", server_type_string) == 0) {
+    return SERVERTYPE_DEV;
+  } else if (strcasecmp("OPERATE", server_type_string) == 0) {
+    return SERVERTYPE_OPERATE;
+  } else if (strcasecmp("WORKING", server_type_string) == 0) {
+    return SERVERTYPE_WORKING;
+  } else if (strcasecmp("DUMMY", server_type_string) == 0) {
+    return SERVERTYPE_DUMMY;
+  } else
+    return SERVERTYPE_NONE;
+}
+
+void BillingManager::Init() {
+  billing_api_handle_ = OpenBillingApi();
+  if (billing_api_handle_ == nullptr) {
+    LOG_ERROR("Fail to open billing api");
+  }
+
+  int init_billing_api = InitBillingApi(billing_api_handle_);
+  LOG_INFO("init_billing_api:%d", init_billing_api);
+  if (init_billing_api == 0) {
+    LOG_ERROR(" Fail to init billing api");
+  }
+}
+
+bool BillingManager::GetProductList(const flutter::EncodableValue *args) {
+  LOG_INFO("start get product list");
+  std::string app_id;
+  std::string country_code;
+  std::string item_type;
+  int page_size;
+  int page_num;
+  std::string check_value;
+  std::string server_type;
+  flutter::EncodableMap encodables = std::get<flutter::EncodableMap>(*args);
+  flutter::EncodableValue &app_id_value =
+      encodables[flutter::EncodableValue("appId")];
+  app_id = std::get<std::string>(app_id_value);
+  flutter::EncodableValue &country_code_value =
+      encodables[flutter::EncodableValue("countryCode")];
+  country_code = std::get<std::string>(country_code_value);
+  flutter::EncodableValue &item_type_value =
+      encodables[flutter::EncodableValue("itemType")];
+  item_type = std::get<std::string>(item_type_value);
+  flutter::EncodableValue &page_size_value =
+      encodables[flutter::EncodableValue("pageSize")];
+  page_size = std::get<int>(page_size_value);
+  flutter::EncodableValue &page_num_value =
+      encodables[flutter::EncodableValue("pageNum")];
+  page_num = std::get<int>(page_num_value);
+  flutter::EncodableValue &check_value_value =
+      encodables[flutter::EncodableValue("checkValue")];
+  check_value = std::get<std::string>(check_value_value);
+  flutter::EncodableValue &server_type_value =
+      encodables[flutter::EncodableValue("serverType")];
+  server_type = std::get<std::string>(server_type_value);
+
+  bool bRet = service_billing_get_products_list(
+      app_id.c_str(), country_code.c_str(), page_size, page_num,
+      check_value.c_str(), ConvertServerType(server_type.c_str()), OnProducts,
+      (void *)this);
+  if (!bRet) {
+    LOG_ERROR("service_billing_get_products_list failed");
+    return false;
+  }
+  return true;
+}
+
+bool BillingManager::GetPurchaseList(const flutter::EncodableValue *args) {
+  LOG_INFO("start get purchase list");
```

Comment:
I think it's better to set 'const flutter::EncodableMap args' as parameter. Otherwise args may be null or not EncodableMap.

Reply:
fixed